pkg/cloud-init/cloud-init.go

```diff
@@ -556,9 +556,9 @@ func GenerateLocalData(vmiName string, namespace string, data *CloudInitData) er
 	if err != nil {
 		return err
 	}
-	diskutils.RemoveFile(metaFile)
-	diskutils.RemoveFile(userFile)
-	diskutils.RemoveFile(networkFile)
+	diskutils.RemoveAllAndLogFailure(metaFile)
```

Comment:
@AlonaKaplan it looks to me  like these calls should actually do something with this error. Possibly fail and block the cleanup until it works again. Could you have a look?

Reply:
Ah sorry, I meant @davidvossel

Reply:
> Possibly fail and block the cleanup until it works again. Could you have a look

yeah. just return the error and handle it the normal way. It was a mistake not to handle the error correctly.

Reply:
@davidvossel let me see if I understand you. Do you suggest that I use `os.RemoveAll` and check  the returned error instead of silently ignoring it with `diskutils.RemoveAllAndLogFailure` ?

I agree that it would be the nobler thing to do.

Should I eliminate `diskutils.RemoveAllAndLogFailure` altogether?

Reply:
You still may want a helper which does not recursively delete directories but can handle non existing files. Like `func removeFileIfExists() error`.

Reply:
> You still may want a helper which does not recursively delete directories but can handle non existing files. Like func removeFileIfExists() error.

I'm not sure. If the only thing you care about when you call `removeFileIfExists(x)` is to ensure that there `x` does not exist, why would you want to handle the case where `x` is a directory with a few files? Either you know exactly that `x` is there and you want to remove it (that's `os.Remove`), or you don't know and don't care and just want nothing to be there (that's os.RemoveAll)

Reply:
when syncrhonizing in eventually consistend envs, we try to deal with situations where previous tries of a step only partially succeeded. We just want to start fresh if something went wrong. But we do not want to recursively directories if we want a file to be removed. As you can see in the code, the purpose of it is removing files. Wile you complain about the naming of  the function, the original function name makes that clear already.

Reply:
Ok. Would you mind if I change it to `removeFilesIfExist(filenames ...string)` to make it simpler to start afresh, yet handle errors if they pop up?

Reply:
if you mean with "yet handle errors if they pop up" that it returns an error too: `func removeFilesIfExist(filenames ...string) error`, sure.

Reply:
@rmohr As we delete specific files and not directories, I think that using os.RemoveAll(filename) will do what you want. Any reason we need special helper function for that ?

Reply:
@ezrasilvera  `RemoveAll` is the equivalent of `rm -rf`. I would consider the helper to be more targeted and more defensive.

Here the docs:

```
RemoveAll removes path and any children it contains. It removes everything it can but returns the first error it encounters. If the path does not exist, RemoveAll returns nil (no error). If there is an error, it will be of type *PathError.
```

Reply:
@rmohr correct, however as I pointed out if the parameter is a file and not directory the recursivness doesn't have any impact (basically `rm -f`  in your example). I agree this is an abuse of the command but it allows you to avoid the "non existent"  error check.

Reply:
What is the problem with playing save and increase the chance of getting errors in CI in case something wrong is passed to it by accident?

Reply:
No problem at all :-)   I just don't like to add redundant utility functions if we can avoid them.  I guess if you want to defend against the "path is a directory" case, you are correct.

Reply:
I would call it the "path is a directory and I delete it by accident recursively and do not even get errors" case. Yes :)

Reply:
error addressed in the third commit of this PR.